ucte/ucte-converter/src/main/java/com/powsybl/ucte/converter/UcteExporter.java

```diff
@@ -0,0 +1,1072 @@
+     * @return The type of the UcteAngleRegulation
+     */
+    private UcteAngleRegulationType findRegulationType(TwoWindingsTransformer twoWindingsTransformer) {
+        if (isSymm(twoWindingsTransformer)) {
+            return UcteAngleRegulationType.SYMM;
+        } else {
+            return UcteAngleRegulationType.ASYM;
+        }
+    }
+
+    private boolean isSymm(TwoWindingsTransformer twoWindingsTransformer) {
+        for (int i = twoWindingsTransformer.getPhaseTapChanger().getLowTapPosition();
+             i < twoWindingsTransformer.getPhaseTapChanger().getHighTapPosition(); i++) {
+            if (twoWindingsTransformer.getPhaseTapChanger().getStep(i).getRho() != 1) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    UcteNodeCode createUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
+        UcteNodeCode ucteNodeCode;
+        if (iidmIdToUcteNodeCodeId.containsKey(id)) {
+            return iidmIdToUcteNodeCodeId.get(id);
+        }
+        Optional<UcteNodeCode> optionalUcteNodeCode = parseUcteNodeCode(id);
+        if (optionalUcteNodeCode.isPresent()) { // the ID is already an UCTE id
+            iidmIdToUcteNodeCodeId.put(id, optionalUcteNodeCode.get());
+            return optionalUcteNodeCode.get();
+        } else {
+            convertIidmIdToUcteNodeCode(id, voltageLevel, country);
+            ucteNodeCode = iidmIdToUcteNodeCodeId.get(id);
+        }
+        return ucteNodeCode;
+    }
+
+    /**
+     * create a unique UcteNodeCode using the voltage level and the country given and
+     * store it to {@link UcteExporter#iidmIdToUcteNodeCodeId} so we don't need to create it every time
+     * and ensure  its uniqueness
+     *
+     * @param id the orignal id of the iidm component we want to convert
+     * @param voltageLevel the {@link VoltageLevel} of the component we want to convert
+     * @param country the string corresponding to the country where the iidm component is
+     */
+    void convertIidmIdToUcteNodeCode(String id, VoltageLevel voltageLevel, String country) {
+        UcteNodeCode ucteNodeCode = null;
+        String voltageNameOrId = voltageLevel.getName() != null ? voltageLevel.getName() : voltageLevel.getId();
+        int busbar = 'a';
+        do {
+            if (busbar <= 'z') {
+                ucteNodeCode = new UcteNodeCode(
+                        UcteCountryCode.valueOf(country),
+                        voltageNameOrId,
+                        voltageLevelCodeFromIidmVoltage(voltageLevel.getNominalV()),
+                        (char) busbar
+                );
+            } else {
+                LOGGER.warn("There is more than 26 nodes with the the id starting by '{}'", ucteNodeCode);
+                generateUcteNodeCode(id, voltageLevel, country);
+                return;
+            }
+            busbar++;
+        } while (iidmIdToUcteNodeCodeId.values().contains(ucteNodeCode));
```

Comment:
This loop is not OK: it's could not be a do-while if you start by a test. You should convert it to a while loop.

Reply:
If I want this to become a while loop I need to have generated the first ucteNodeCode before the loop, so I'll end up with some code duplication or I need to add as a condition in the while loop `while(ucteNodeCode == null`. I'm not sure it's better than the do while.